### PR TITLE
Update rules of identifier

### PR DESCRIPTION
**Pull request by bidoubiwa**
Fixes: meilisearch/MeiliSearch#416 :

* The document id must contain only A-Z a-z 0-0 and -_. This is to avoid encoding in the url path in the case of a multi-word id.

### CODE REVIEW

File: main_concepts/documents.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 `Documents` are objects composed of fields containing any data.</br>
 A `field` is composed of an `attribute` and its associated data.
 
+
```
Comment (bidoubiwa) [inline main_concepts/documents.md]:
```suggestion
```

File: main_concepts/indexes.md [hunk 2]
```diff
@@ -101,11 +101,18 @@ If two documents added at the same time have the same ID, MeiliSearch will only
 
 If a document is added when there is already a document with the same ID in MeiliSearch, it will be updated.
 
+::: warning
+**The identifier is mandatory**, and must [follow formatting rules](/main_concepts/documents.md#documents-identifiers).
+:::
+
+
```
Comment (bidoubiwa) [inline main_concepts/indexes.md]:
```suggestion
```

File: main_concepts/indexes.md [hunk 3]
```diff
@@ -101,11 +101,18 @@ If two documents added at the same time have the same ID, MeiliSearch will only
...
 ::: tip
 Documents identifiers are always converted into strings. Only strings and integers are valid identifiers.
 It means that it is, for example, forbidden to use arrays or objects as an identifier.
 :::
 
+
+
```
Comment (bidoubiwa) [inline main_concepts/indexes.md]:
```suggestion
```

File: main_concepts/documents.md [hunk 4]
```diff
@@ -13,15 +13,35 @@ A `field` is composed of an `attribute` and its associated data.
...
-The **fields** are the combination of attributes and data (i.e. `"title": "Interstellar"`).
+The **fields** are the combination of attributes and data (i.e., `"title": "Interstellar"`).
 
 ## Documents structure
 
 A document is added to MeiliSearch in `JSON format`.<br/>
-Each documents fields should correspond to an attribute in the schema to be taken into account.
+Each field in a document should correspond to a field defined in the [schema](/main_concepts/indexes.md#schema-definition) to be taken into account.
+
+A **document must contain** [one identifier field](/main_concepts/documents.md#documents-identifiers) to be indexed in meiliSearch.
```
Comment (curquiza) [inline main_concepts/documents.md]:
```suggestion
A **document must contain** [one identifier field](/main_concepts/documents.md#documents-identifiers) to be indexed in MeiliSearch.
```

File: main_concepts/documents.md [hunk 5]
```diff
@@ -13,15 +13,35 @@ A `field` is composed of an `attribute` and its associated data.
...
 
-When using the [route to add new documents](/references/documents.md#add-or-update-documents) all documents should be sent in an array. And this, even if there is only one document.
+When using the [route to add new documents](/references/documents.md#add-or-update-documents), all documents should be sent in an array. And this, even if there is only one document.
+
+## Documents identifiers
+
+To be indexed by MeiliSearch, a document must have an **identifier**. A document without an identifier will be ignored when added to MeiliSearch.
+
+The identifier attribute name can be anything (unless the [schema is inferred when adding documents](/main_concepts/indexes.md#inferred-schema)).
+
```
Comment (curquiza) [inline main_concepts/documents.md]:
```suggestion
```

File: main_concepts/documents.md [hunk 6]
```diff
@@ -13,15 +13,35 @@ A `field` is composed of an `attribute` and its associated data.
...
+
+The identifier value must contain only `A-Z a-z 0-0` and `-_`.
```
Comment (curquiza) [inline main_concepts/documents.md]: I find this confusing to read "The identifier attribute name can be anything" and then " The identifier value must contain only `A-Z a-z 0-0` and `-_`. "

File: main_concepts/indexes.md [hunk 7]
```diff
@@ -101,11 +101,16 @@ If two documents added at the same time have the same ID, MeiliSearch will only
...
+::: warning
+**The identifier is mandatory**, and must [follow formatting rules](/main_concepts/documents.md#documents-identifiers).
+:::
+
 ::: tip
 Documents identifiers are always converted into strings. Only strings and integers are valid identifiers.
 It means that it is, for example, forbidden to use arrays or objects as an identifier.
 :::
 
+
```
Comment (curquiza) [inline main_concepts/indexes.md]:
```suggestion
```

**Pull request merged**